src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1393,8 +1395,26 @@ export default class AircraftModel {
                 break;
 
             case FLIGHT_PHASE.APPROACH: {
-                if (this.positionModel.distanceToPosition(this.mcp.nav1Datum) < AIRPORT_CONSTANTS.FINAL_APPROACH_FIX_DISTANCE_NM) {
-                    this.setFlightPhase(FLIGHT_PHASE.LANDING);
+                const datum = this.mcp.nav1Datum;
+                if (this.positionModel.distanceToPosition(datum) < AIRPORT_CONSTANTS.FINAL_APPROACH_FIX_DISTANCE_NM) {
+                    const angleAwayFromLocalizer = this.mcp.course - this.positionModel.bearingToPosition(datum);
+                    const course = this.mcp.course;
+                    const courseOffset = getOffset(this, datum.relativePosition, course);
+                    const lateralDistanceFromCourseNm = nm(courseOffset[0]);
+
+                    const isAlignedWithCourse = abs(lateralDistanceFromCourseNm) <= PERFORMANCE.MAXIMUM_DISTANCE_CONSIDERED_ESTABLISHED_ON_APPROACH_COURSE_NM;
+                    const isCourseStabilised = abs(angleAwayFromLocalizer) < PERFORMANCE.MAXIMUM_ANGLE_CONSIDERED_ESTABLISHED_ON_APPROACH_COURSE;
+                    const isOnGlideslope = abs(this.altitude - this.target.altitude) < 100;
+
+                    if (isCourseStabilised && isOnGlideslope && isAlignedWithCourse) {
```

Comment:
Move this inner if/else to a new method and remove the else within that method.

Reply:
- `const isAlignedWithCourse` is already covered by a method `RunwayModel.isOnApproachCourse()`
`const isCourseStabilized` is already covered by a method `RunwayModel.isOnCorrectApproachHeading()`

Reworked this whole thing to make use of functions like those (and add ones we should have had), etc. This whole section reads really nicely now I think.